credentials/alts/internal/testutil/testutil.go

```diff
@@ -67,20 +68,36 @@ func (s *Stats) Reset() {
 // testConn mimics a net.Conn to the peer.
 type testConn struct {
 	net.Conn
-	in  *bytes.Buffer
-	out *bytes.Buffer
+	in          *bytes.Buffer
+	out         *bytes.Buffer
+	readLatency time.Duration
 }
 
 // NewTestConn creates a new instance of testConn object.
 func NewTestConn(in *bytes.Buffer, out *bytes.Buffer) net.Conn {
 	return &testConn{
-		in:  in,
-		out: out,
+		in:          in,
+		out:         out,
+		readLatency: time.Duration(0),
+	}
+}
+
+// NewTestConnWithReadLatency creates a new instance of testConn object that
+// pauses for readLatency before any call to Read() returns.
+func NewTestConnWithReadLatency(
+	in *bytes.Buffer,
+	out *bytes.Buffer,
+	readLatency time.Duration) net.Conn {
```

Comment:
Please don't split up lines this way. Go does not enforce any line length limits. See: https://google.github.io/styleguide/go/guide#line-length

Reply:
Done.